src/main/kotlin/com/cjbooms/fabrikt/generators/model/JacksonModelGenerator.kt

```diff
@@ -462,22 +465,8 @@ class JacksonModelGenerator(
         extensions: Map<String, Any>,
         oneOfInterfaces: Set<SchemaInfo>,
     ): TypeSpec {
-        val filteredProperties = if (oneOfInterfaces.size == 1) {
```

Comment:
This is the logic in question. I'm not sure of the intent here, but perhaps @pschichtel can weigh in before I revert it

Reply:
the logic is:

if:
  - the super interface' model defines a discriminator field
  - the discriminator mappings map to the current model exactly once
then:
  - remove the discriminator field, since it must always be the same value given by the jackson type info field.